llama2.mojo

```diff
@@ -370,14 +370,16 @@ struct Config:
     var shared_weights: Bool
 
     fn __init__(inout self, fileName: String, print_config: Bool) raises:
-        var f = open(fileName, "r")
-        # reading 7 vars of type DType.int32 from the file
-        var bytes_of_config_params = NUM_CONFIG_INT * sizeof[DType.int32]()
-        # config_data_raw id Tensor[DType.int8] with bytes_of_config_params elements
-        var config_data_raw = f.read_bytes(bytes_of_config_params)
-        f.close()
+        var bytes_of_config_params: Int
+        var config_data_raw: String
+        with open(fileName, "rb") as f:
+            # reading 7 vars of type DType.int32 from the file
+            bytes_of_config_params = NUM_CONFIG_INT * sizeof[DType.int32]()
+            # config_data_raw id Tensor[DType.int8] with bytes_of_config_params elements
+            config_data_raw = f.read(bytes_of_config_params)
+
         # correct Tensor type and shape for easy reading, without copying data
-        var int32_ptr = DTypePointer[DType.int8](config_data_raw.steal_data().value).bitcast[DType.int32]()
+        var int32_ptr = config_data_raw._steal_ptr().bitcast[DType.int32]()
```

Comment:
I would not return back the `__steal_ptr()` , since it may be removed from the future releases. Do you see any benefits of `__steal_ptr()` ?

Reply:
Not really, I just used it because was easier to read.